pynicotine/users.py

```diff
@@ -28,14 +28,16 @@
 
 class WatchedUser:
 
-    __slots__ = ("username", "upload_speed", "files", "folders")
+    __slots__ = ("username", "upload_speed", "files", "folders", "modes", "implicit")
 
     def __init__(self, username):
 
         self.username = username
         self.upload_speed = None
         self.files = None
         self.folders = None
+        self.modes = set()
```

Comment:
String matching to identify different modes isn't very nice.

Consider defining a new class such as `WatchMode()` for this purpose.

Reply:
It doesn't feel right to make the users module responsible for knowing every mode in the application. Ideally you could call watch_user() from anywhere (say, a plugin) without being required to add a new mode here.

Reply:
Perhaps get a value without needing to pass an argument with something in https://docs.python.org/3/library/inspect.html#module-inspect